src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -90,6 +92,20 @@ static bool exclusive = false;
 #endif
 #define htonll(a) ntohll(a)
 
+static int do_unmap();
+
+static void handle_signal(int signum)
+{
+  assert(signum == SIGINT || signum == SIGTERM);
+  derr << "*** Got signal " << sig_str(signum) << " ***" << dendl;
+  dout(20) << __func__ << ": " << "sending NBD_DISCONNECT" << dendl;
+  if (ioctl(nbd, NBD_DISCONNECT) < 0) {
+    derr << "rbd-nbd: disconnect failed" << cpp_error(errno) << std::endl;
+  } else {
+    dout(20) << __func__ << ": " << "disconnected" << dendl;
+  }
```

Comment:
@yangdongsheng Thanks for your analysis. 
\(1\) For your first question, I think there are two different kinds of perspective: from rbd-nbd, or from nbd driver. I agree with Trociny because in this way, the user doesn't need to care the detailed implementation in nbd driver. @trociny, please correct me.

\(2\) From your example, I believe the second disconnect: "disconnect in sig_handler" in Thread1 will report error: 
rbd-nbd: disconnect failed: " << cpp_error(errno) , and we also dump error code, so I don't think it is a big issue.